templates/ingress-daemon-set.yaml

```diff
@@ -4,6 +4,7 @@ metadata:
   name: ingress-nginx-{{ juju_application }}
   labels:
     cdk-{{ juju_application }}-ingress: "true"
+    cdk-ingress: "true"
```

Comment:
Why add a new label when there was already a `cdk-{{ juju_application }}-ingress` label?

Reply:
It's so kubernetes-master can select it here: https://github.com/charmed-kubernetes/charm-kubernetes-master/pull/34/files#diff-bf8eaea470605940d2c1bc8d1a7ecd39R2448

The `cdk-{{ juju_application }}-ingress` label is problematic because you need to know the application name in order to select it. It's usually `kubernetes-worker`, but not always.

So I had two options here:
1. Make each kubernetes-worker application responsible for restarting its own ingress controller (which admittedly does make sense), or
2. Add a new cdk-ingress label so that kubernetes-master can restart them along with the other addons (which is quicker and easier)

I went with option 2.

Reply:
Technically, the master would have the application name via the relation, but I like the shorter label better anyway.  +1

Reply:
I dropped the `cdk-ingress` label in favor of a new label, `cdk-restart-on-ca-change`, which is applied to all Deployments/DaemonSets/StatefulSets deployed by CDK charms or the cdk-addons snap.

The motivation for this is that I wanted to be able to add Calico addons to the addon restart code without introducing yet another label.